timeline-chart/src/layer/time-graph-chart.ts

```diff
@@ -497,51 +500,83 @@ export class TimeGraphChart extends TimeGraphChartLayer {
             // Only if we are updating all rows and not increasing vertical height.  See: https://github.com/eclipse-cdt-cloud/theia-trace-extension/pull/832#issuecomment-1259902534.
             const allRowsUpdated = rowIds.length === visibleRowIds.length;
             const worldRange = allRowsUpdated ? this.unitController.updateWorldRangeFromViewRange() : this.unitController.worldRange;
-            const request = { worldRange, resolution, rowIds };
-            if (isEqual(request, this.ongoingRequest)) {
-                // request ignored because equal to ongoing request
-                return;
+            const additionalParams: { [key: string]: any } = {};
+            if (this._filterExpressionsMap) {
+                additionalParams['filter_query_parameters'] = {'filter_expressions_map': this._filterExpressionsMap, 'strategy': fullSearch ? 'DEEP' : 'SAMPLED'};
```

Comment:
@hriday-panchasara I tried to come-up with a proposal that makes this less coupled to the `theia-trace-extension` implementation. However, it's not straight forward due the need to pass the strategy back. If you meanwhile have found a way please let me know.

So, I opened a ticket so that we can address the tight coupling in a later effort #256. My suggestion going forward is to keep this implementation as proposed in the PR (you can add a comment to the new APIs about #256). Afterwards, we can work on the refactoring of the implementation when fixing #256. We can then take our time to come-up with a better, decoupled API and implementation. WDYT?

Reply:
Thanks for the suggestion. I'm also struggling to take out the `filter_query_parameters`, mostly because the current implementation is tied with the changes brought by #180 & #205. Therefore, decoupling from this point would affect current functionality. My suggestion also would be to use #256 to roll back and find a solution that satisfies all the functional requirements and performance optimizations that exist within this library.